aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -49,6 +49,34 @@ The following timeouts and rate limits aren't enforced when a debugger is attach
 * <xref:Microsoft.AspNetCore.Server.Kestrel.Core.Features.IHttpMinRequestBodyDataRateFeature>
 * <xref:Microsoft.AspNetCore.Server.Kestrel.Core.Features.IHttpMinResponseDataRateFeature>
 
+## Shutdown
+
+When an ASP.NET Core app receives a shutdown signal, it goes through a series of steps to gracefully shut down the server:
+
+* The Host receives a shutdown signal, for example, from `CTL+C`, [StopAsync](/aspnet/core/fundamentals/host/hosted-services#stopasync), etc.
+* [IHostApplicationLifetime.ApplicationStopping](xref:Microsoft.Extensions.Hosting.IHostApplicationLifetime.ApplicationStopping) is signaled to notify the app. Long running operations should subscribe to this event.
+* The Host calls [IServer.StopAsync](xref:Microsoft.AspNetCore.Hosting.Server.IServer.StopAsync%2A) with a configurable shutdown timeout. The default shutdown is 30 seconds.
+* Kestrel (and [Http.Sys](/aspnet/core/fundamentals/servers/#kestrel-vs-httpsys)):
+  * Close their port bindings and stop accepting new connections.
+  * Signal current connections to stop processing new requests. For HTTP/2 and HTTP/3 this involves sending a preliminary GoAway message to the client. For HTTP/1.1, requests are processed in order so it stops that connection loop.
+    * IIS is a little different, it rejects new requests with a [503](https://developer.mozilla.org/docs/Web/HTTP/Status/503) status.
```

Comment:
The fact that this is describing multiple servers behaviors makes me think it belongs in a different doc. Maybe this whole thing should go in hosting?